### ISSUE TITLE
Font loading does not handle missing system_profiler command from PATH

### ISSUE THREAD

**Issue by lassandroan** [pygame/pygame#2549]
**Environment:**

- **Operating system**: macOS Mojave 10.14.6
- **Python version**: 3.9.1
- **SDL version**: 2.0.14
- **PyGame version**: 2.0.1
- **Relevant hardware**: n/a

**Current behavior:**

If you run in an environment where your path may be incorrect and `system_profiler` is not found, the font loading module of pygame will crash with a seemingly unrelated exception (`xml.etree.ElementTree.ParseError`).

In my case I'm running the `fish` shell on a computer I bought and set up recently, and `/usr/sbin/` was not in my path. Adding it in fixed the problem, but I had to debug the pygame library to find out what was happening.

**Expected behavior:**

My expectation would be that if the subprocess for loading fonts fails to run the command, that an error detailing as such will be raised. This is the offending function:

```python
def _system_profiler_darwin():
    fonts = {}
    flout, _ = subprocess.Popen(
        ' '.join(['system_profiler', '-xml', 'SPFontsDataType']),
        shell=True,
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE,
        close_fds=True
    ).communicate()

    for font_node in ET.fromstring(flout).iterfind('./array/dict/array/dict'):
        _add_font_paths(font_node.iter("*"), fonts)

    return fonts
```

stderr is assigned to `_` and contains information regarding the error `Unknown command: system_profiler`, but because the error isn't accounted for the code continues. `flout`, which is now an empty binary string `b''`, is then passed to `fromstring()` which then produces the exception shown in my screenshot.

**Screenshots**

![2021-04-03 23 42 38](https://user-images.githubusercontent.com/9328186/113498849-51ba2d00-94d6-11eb-988b-148ba285644c.jpg)

**Steps to reproduce:**

This code will immediately fail if your path is missing `/usr/sbin/`, since pygame can't find `system_profiler`:

```python
import pygame
pygame.init()
font = pygame.font.SysFont('Arial', 12)
```

**Test code**

Listed above

**Stack trace/error output/other error logs**

N/A

**Comment by Starbuck5** [comment]
Thanks for the report.

Perhaps the expected behavior here should be to emit a warning.
That way the user can see the problem, but the code won't crash, since the documentation specifies:

> If a suitable system font is not found this will fall back on loading the default pygame font.

**Comment by Starbuck5** [comment]
I found a simple fix for this, but I would prefer to fix it a different way, which would also fix #2553

For reference sake, the call to system_profiler can be directly to to `usr/sbin/system_profiler`, that way it doesn't matter if it is in the path or not.
